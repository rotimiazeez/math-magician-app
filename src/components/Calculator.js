import '../App.css';
import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.handleEvent = this.handleEvent.bind(this);
  }

  handleEvent = (e) => {
    e.preventDefault();
    this.setState((prev) => calculate(prev, e.target.name));
  }

  render() {
    const { total, next, operation } = this.state;
    const showTotal = () => {
      if (total === 0 && next === null) {
        return 0;
      }

      if (total !== 0 && total !== null) {
        return total;
      }
      return '';
    };
    return (
      <div className="container">
        <form>
          <input type="text" id="result" value={(showTotal()) + (operation || '') + (next || '')} disabled />
        </form>

        <div className="keypad">
          <button type="button" onClick={this.handleEvent} name="AC">AC</button>
          <button type="button" onClick={this.handleEvent} name="+/-">+/-</button>
          <button type="button" onClick={this.handleEvent} name="%">%</button>
          <button type="button" onClick={this.handleEvent} name="รท" className="highlight">รท</button>
          <button type="button" onClick={this.handleEvent} name="7">7</button>
          <button type="button" onClick={this.handleEvent} name="8">8</button>
          <button type="button" onClick={this.handleEvent} name="9">9</button>
          <button type="button" onClick={this.handleEvent} name="x" className="highlight">x</button>
          <button type="button" onClick={this.handleEvent} name="4">4</button>
          <button type="button" onClick={this.handleEvent} name="5">5</button>
          <button type="button" onClick={this.handleEvent} name="6">6</button>
          <button type="button" onClick={this.handleEvent} name="-" className="highlight">-</button>
          <button type="button" onClick={this.handleEvent} name="1">1</button>
          <button type="button" onClick={this.handleEvent} name="2">2</button>
          <button type="button" onClick={this.handleEvent} name="3">3</button>
          <button type="button" onClick={this.handleEvent} name="+" className="highlight">+</button>
          <button type="button" onClick={this.handleEvent} name="0" className="bottom">0</button>
          <button type="button" onClick={this.handleEvent} name="." className="bottomII">.</button>
          <button type="button" onClick={this.handleEvent} name="=" className="highlight bottomIII">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
