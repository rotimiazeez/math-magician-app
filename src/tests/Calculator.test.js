/* eslint-disable no-restricted-syntax */
import React from 'react';
import {
  render, screen, cleanup, fireEvent,
} from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('calculator testing', () => {
  test('2 x 3', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));

    expect(screen.getByText('6')).toMatchSnapshot();
  });

  test('calculator contains all buttons', () => {
    const calcButtons = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];

    render(
      <Calculator />,
    );

    for (const btn of calcButtons) {
      expect(screen.getByText(btn)).toBeInTheDocument();
    }
  });

  test('clicking the button toggles the number', () => {
    render(<Calculator />);
    const button = screen.getByText('1');
    // fireEvent.click(button);
    expect(button).toBeInTheDocument();
  });
});

afterEach(cleanup);
