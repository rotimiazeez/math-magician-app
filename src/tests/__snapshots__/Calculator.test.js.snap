// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator testing 2 x 3 1`] = `
<button
  name="6"
  type="button"
>
  6
</button>
`;
